Source/PersonalityInsightsV3/PersonalityInsights.swift

```diff
@@ -0,0 +1,274 @@
+    private func dataToError(data: Data) -> NSError? {
+        do {
+            let json = try JSON(data: data)
+            let code = try json.getInt(at: "code")
+            let error = try json.getString(at: "error")
+            let help = try json.getString(at: "help")
+            let userInfo = [
+                NSLocalizedFailureReasonErrorKey: error,
+                NSLocalizedRecoverySuggestionErrorKey: help
+            ]
+            return NSError(domain: domain, code: code, userInfo: userInfo)
+        } catch {
+            return nil
+        }
+    }
+
+    /**
+     Analyze text to generate a personality profile.
+     
+      - parameter version: The date in the form "YYYY-MM-DD" that will specify a version no later than the date provided.
+     - parameter text: The text to analyze.
+     - parameter acceptLanguage: The desired language of the response.
+     - parameter contentLanguage: The language of the text being analyzed.
+     - parameter rawScores: If true, then a raw score for each characteristic is returned in
+        addition to a normalized score. Raw scores are not compared with a sample population.
+        A raw sampling error for each characteristic is also returned.
+     - parameter consumptionPreferences: If true, then information inferred about consumption preferences is returned in addition to the results.
+     - parameter failure: A function executed if an error occurs.
+     - parameter success: A function executed with the personality profile.
+     */
+    public func getProfile(
+        version: String,
+        fromText text: String,
+        acceptLanguage: String? = nil,
+        contentLanguage: String? = nil,
+        rawScores: Bool? = nil,
+        consumptionPreferences: Bool? = nil,
+        failure: ((Error) -> Void)? = nil,
+        success: @escaping (Profile) -> Void)
+    {
+        guard let content = text.data(using: String.Encoding.utf8) else {
+            let failureReason = "Text could not be encoded to NSData with NSUTF8StringEncoding."
+            let userInfo = [NSLocalizedFailureReasonErrorKey: failureReason]
+            let error = NSError(domain: domain, code: 0, userInfo: userInfo)
+            failure?(error)
+            return
+        }
+
+        getProfile(
+            version: version,
+            fromContent: content,
+            withType: "text/plain",
+            acceptLanguage: acceptLanguage,
+            contentLanguage: contentLanguage,
+            rawScores: rawScores,
+            consumptionPreferences: consumptionPreferences,
+            failure: failure,
+            success: success
+        )
+    }
+
+    /**
+     Analyze the text of a webpage to generate a personality profile.
+     The HTML tags are stripped before the text is analyzed.
+     
+     - parameter version: The date in the form "YYYY-MM-DD" that will specify a version no later than the date provided.
+     - parameter html: The webpage that contains text to analyze.
```

Comment:
It would be nice to improve this comment. As a user, I think it be confusing whether this should be the URL of a webpage or the body of an HTML document. Maybe we could change it to this instead:

`fromHTML: The text of an HTML webpage.`

Though I'm not sure if that's any clearer...